tomviz/ModuleRuler.cxx

```diff
@@ -150,8 +150,10 @@ bool ModuleRuler::setVisibility(bool val)
 {
   vtkSMPropertyHelper(m_Representation, "Visibility").Set(val ? 1 : 0);
   m_Representation->UpdateVTKObjects();
-  if (m_Widget) {
+  if (!val || m_showLine) {
```

Comment:
Yeah, calling setWidgetVisible causes the widget to emit a signal and our signal handler overrides the m_showLine variable.  In this case we don't want the member variable being overridden, so we have to cache it to a local variable and restore it.  I'll add a comment.

The null check... I guess I can put it back.  It didn't matter in any of the cases I tested (and I tried to be thorough since I was removing a null check).